modules/raft/tech-notes/rebalance.md

```diff
@@ -0,0 +1,71 @@
+## Introduction
+Since the last rebalance design we made some significant decisions and architecture updates:
+- Transaction protocol introduce the new cluster/group wide roles like Tracker (Placement Driver), LeaseHolder (Primary replica) and etc. (see [Transaction Protocol](https://cwiki.apache.org/confluence/display/IGNITE/IEP-91%3A+Transaction+protocol))
+- The protocol of replication itself will be extended to the pluggable abstraction, instead of RAFT-only one. (TODO: add the document link, when it will be ready to share).
+- New distribution zones layer were introduced (see [Distribution Zones](https://cwiki.apache.org/confluence/display/IGNITE/IEP-97%3A+Distribution+Zones)). So, the assignment property is not the part of table configuration anymore.
+
+These changes incline us to the thoughts, that we need to revise the current rebalance flow, because it doesn't suite to the new architecture anymore in general and doesn't use the power of new abstractions on the other side.
+
+## Rebalance triggers
+The simplest way to start the journey to the new design: look at the real cases and try to draw the whole picture.
+
+We still has the number of triggers, which trigger a rebalance:
+- Change the number of replicas for any distribution zone.
+- Change the number of partitions for any distribution zone.
+- Change the distribution zone data nodes composition.
+
+Let's take the first one to draw the whole rebalance picture.
+## Change the number of replicas
```

Comment:
This document structure implies that for now we are considering only the first trigger, is it so?

Reply:
Yep, during this doc we want to show the rebalance process core at the moment. But, I suppose we will extend it in the future.